discovery/marathon/marathon.go

```diff
@@ -201,20 +204,29 @@ type AppList struct {
 }
 
 // AppListClient defines a function that can be used to get an application list from marathon.
-type AppListClient func(client *http.Client, url string) (*AppList, error)
+type AppListClient func(client *http.Client, url, token string) (*AppList, error)
 
 // fetchApps requests a list of applications from a marathon server.
-func fetchApps(client *http.Client, url string) (*AppList, error) {
+func fetchApps(client *http.Client, url, token string) (*AppList, error) {
 	resp, err := client.Get(url)
+	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		return nil, err
 	}
 
+	if token != "" {
+		req.Header.Add("Authorization", "token="+token)
+	}
+	resp, errl := client.Do(req)
```

Comment:
`err` is already defined in that scope, wouldn't that will become a compilation error ?